### PR TITLE
IA-1669 IA-1731 attempt to upgrade swagger ui to openAPI 3

### PR DESCRIPTION
**Pull request by Qi77Qi**
Still need to do https://broadworkbench.atlassian.net/browse/IA-1744.

Screen Shot 2020-03-09 at 11 36 32 PM

Screen Shot 2020-03-09 at 10 58 36 PM

Screen Shot 2020-03-09 at 11 35 09 PM

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -1,16 +1,13 @@
-swagger: '2.0'
```
Comment (Qi77Qi) [inline http/src/main/resources/swagger/api-docs.yaml:1]: this conversion was done via a tool that auto converts swagger 2 to openAPI 3

Comment (rtitle) [reply http/src/main/resources/swagger/api-docs.yaml:1]: Can you run it though an openapi validator tool to be safe?

Also should we confirm this doesn't affect AoU? Might be good to run a test of their code-gen if possible.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/SwaggerRoutes.scala [hunk 2]
```diff
@@ -32,40 +27,43 @@ class SwaggerRoutes(swaggerConfig: SwaggerConfig) {
...
           original.utf8String
-            .replace("your-client-id", swaggerConfig.googleClientId)
-            .replace("your-realms", swaggerConfig.realm)
-            .replace("your-app-name", swaggerConfig.realm)
-            .replace(scopeSeparator(","), scopeSeparator(" "))
-            .replace("jsonEditor: false,", "jsonEditor: false," + swaggerOptions)
-            .replace("""url = "http://petstore.swagger.io/v2/swagger.json";""", "url = '/api-docs.yaml';")
+            .replace("""url: "https://petstore.swagger.io/v2/swagger.json"""", "url: '/api-docs.yaml'")
+            .replace("""layout: "StandaloneLayout"""", s"""layout: "StandaloneLayout", $swaggerOptions""")
+            .replace("window.ui = ui", s"""ui.initOAuth({
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/SwaggerRoutes.scala:52]: So OAuth works with these params?

**Review by rtitle — APPROVED** [review]
LGTM, thanks for doing this 👍

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/SwaggerRoutes.scala (hunk 2 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/SwaggerRoutes.scala:52]: yes

File: http/src/main/resources/swagger/api-docs.yaml (hunk 1 above)
Comment (Qi77Qi) [reply http/src/main/resources/swagger/api-docs.yaml:1]: I opened it in vscode where I have swagger/openapi 3 linter installed, and it looks fine. I slacked @calbach  and @jmthibault79 a few days ago and they confirmed there's no concerns.

@calbach @jmthibault79 can you guys run some codegen that you guys use against the new spec? or let me know which codegen you guys are using, and I can validate

Comment (kyuksel) [reply http/src/main/resources/swagger/api-docs.yaml:1]: No errors on http://editor.swagger.io 👍

File: http/src/main/resources/swagger/api-docs.yaml [hunk 3]
```diff
@@ -785,552 +939,655 @@ paths:
...
-        type: string
-        description: |
-          Optional, the machine type determines the number of CPUs and memory for the master node. For example "n1-standard-16"
-          or "n1-highmem-64". To decide which is right for you, see https://cloud.google.com/compute/docs/machine-types
-      masterDiskSize:
-        type: integer
-        description: |
-          Optional, the size in gigabytes of the disk on the master node. Minimum size is 10GB. Note: disk size cannot be
-          decreased in an update request, only increased.
+            decreased in an update request, only increased.
```
Comment (kyuksel) [inline http/src/main/resources/swagger/api-docs.yaml:1593]: Nit: add a new line

**Review by kyuksel — APPROVED** [review]
The page looks nice; thank you!

**Comment by mtmorgan** [comment]
We (Bioconductor) make heavy use of the current API, and would like some lead time to update / validate...

**Pull request merged**